source/Missions/Services/Arena/CoopArenaController.cs

```diff
@@ -190,58 +187,170 @@ private void Handle_JoinInfo(MessagePayload<NetworkMissionJoinInfo> payload)
+                //{
+                //    missionBehavior.OnAgentShootMissile(shooter, shot.WeaponIndex, shot.Position, shot.Velocity, shot.Orientation, shot.HasRigidBody, shot.ForcedMissileIndex);
+                //}
+
+                Mission.Current.AddCustomMissile(shooter, missionWeapon, shot.Position, shot.Velocity, shot.Orientation, missionWeapon.GetWeaponComponentDataForUsage(0).MissileSpeed, missionWeapon.GetWeaponComponentDataForUsage(0).MissileSpeed, true, null, payload.What.MissileIndex);
+
+
+                //MissionWeapon weapon;
+                //if (shooter.Equipment[weaponIndex].CurrentUsageItem.IsRangedWeapon && shooter.Equipment[weaponIndex].CurrentUsageItem.IsConsumable)
+                //{
+                //    weapon = shooter.Equipment[weaponIndex];
+                //}
+                //else
+                //{
+                //    weapon = shooter.Equipment[weaponIndex].AmmoWeapon;
+                //}
+                //weapon.Amount = 1;
+                //WeaponData weaponData = weapon.GetWeaponData(true);
+                //WeaponStatsData[] weaponStatsData = weapon.GetWeaponStatsData();
+                //Vec3 direction = shot.Velocity;
+                //float num = direction.Normalize();
+                //WeaponComponentData currentUsageItem = shooter.Equipment[shooter.GetWieldedItemIndex(Agent.HandIndex.MainHand)].CurrentUsageItem;
+                //num = MissionGameModels.Current.AgentApplyDamageModel.CalculateEffectiveMissileSpeed(shooter, currentUsageItem, ref direction, num);
+                //float num2 = (float)shooter.Equipment[shooter.GetWieldedItemIndex(Agent.HandIndex.MainHand)].GetModifiedMissileSpeedForCurrentUsage();
+
+                //AddMissileAuxMethod.Invoke(
+                //    Mission.Current,
+                //    new object[] {
+                //    shot.ForcedMissileIndex,
+                //    true,
+                //    shooter,
+                //    missionWeapon.GetWeaponData(false),
+                //    missionWeapon.GetWeaponStatsData(),
+                //    0,
+                //    shot.Position,
+                //    shot.Velocity,
+                //    shot.Orientation,
+                //    10,
+                //    10,
+                //    shot.HasRigidBody,
+                //    null,
+                //    true,
+                //    null
+                //    });
+
+            });
 
-            OnAgentShootMissileMethod.Invoke(Mission.Current, new object[] { 
-                agentGroupController.ControlledAgents[shot.AgentGuid], 
-                shot.WeaponIndex, 
-                shot.Position,
-                shot.Velocity, 
-                shot.Orientation, 
-                shot.HasRigidBody, 
-                true, 
-                shot.ForcedMissileIndex });
         }
 
         private void Handler_AgentDeath(MessagePayload<AgentDied> obj)
```

Comment:
Could you move this to it's own class, comment, and create a test for this method? It will also need to be autoactivated.